sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -6,35 +6,40 @@ import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
 import android.widget.FrameLayout
 import androidx.annotation.*
 import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListUpdateCallback
 import com.mapbox.bindgen.Expected
 import com.mapbox.maps.viewannotation.*
 import com.mapbox.maps.viewannotation.ViewAnnotation
 import com.mapbox.maps.viewannotation.ViewAnnotation.Companion.USER_FIXED_DIMENSION
 import com.mapbox.maps.viewannotation.ViewAnnotationVisibility
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.CopyOnWriteArraySet
-import kotlin.collections.HashMap
 
 internal class ViewAnnotationManagerImpl(
-  private val mapView: MapView
+  mapView: MapView
 ) : ViewAnnotationManager, ViewAnnotationPositionsUpdateListener {
 
+  private val annotationsRootView: FrameLayout = FrameLayout(mapView.context).also {
+    it.layoutParams = FrameLayout.LayoutParams(
+      ViewGroup.LayoutParams.MATCH_PARENT,
+      ViewGroup.LayoutParams.MATCH_PARENT,
+    )
+  }
   private val mapboxMap: MapboxMap = mapView.getMapboxMap()
-  private val viewPlugins = mapView.mapController.pluginRegistry.viewPlugins
   private val renderThread = mapView.mapController.renderer.renderThread
 
   init {
+    // place the view annotations above the map (index 0) but below the compass, ruler and other plugin views
+    mapView.addView(annotationsRootView, 1)
```

Comment:
Question, as I'm not sure of the answer without coding it :D, if the user inflates the MapView from an XML which contains other Views, are we sure that we should add annotationsRootView at index 1? Or should we check how many Views the ViewGroup contains, and add it at the end?

Reply:
It's not the xml that user inflates, it's the FrameLayout of the MapView. MapView adds a SurfaceView/TextureView that actually displays map at the index 0 [here](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/main/java/com/mapbox/maps/MapView.kt#L109), then additional views are added on top of it [here](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk/src/main/java/com/mapbox/maps/plugin/MapPluginRegistry.kt#L71), thus before the fix the layout was looking the following way : 
```
FrameLayout MapView : 
 - SurfaceView/TextureView
 - CompassPlugin
 - LogoPlugin
 - ViewAnnotation1
 - ViewAnnotation2
 - ...
 ```

After the fix it becomes : 
```
FrameLayout MapView : 
 - SurfaceView/TextureView
 - CompassPlugin
 - LogoPlugin
 - FrameLayout for view annotations
   - ViewAnnotation1
   - ViewAnnotation2
 - ...
```